src/main/java/classify/user/InputParsing.java

```diff
@@ -95,23 +95,46 @@ public static void parseUserCommand(String[] userCommand, ArrayList<Student> mas
 
     /**
      * Lets the user check view a list of students with that corresponding subject.
-     * Will only exit from the command if the user types the command exit.
+     * If the user types view_subject [subject], it will only generate the list of students with that subject,
+     * then exit.
+     * If the user types view_subject, the user can continuously view all students that
+     * have that subject, until they exit the command.
      *
      * @param masterStudentList The list of all students.
      * @param in                The user's input.
      */
-    private static void handleViewSubjectCommand(ArrayList<Student> masterStudentList, Scanner in) {
+    private static void handleViewSubjectCommand(ArrayList<Student> masterStudentList, Scanner in, String subject) {
+        if (subject != null && !subject.isEmpty()) {
+            viewStudentsBySubject(masterStudentList, subject);
+        } else {
+            findStudentsWithSubject(masterStudentList, in);
+        }
+    }
+
+    /**
+     * Finds students with the specified subject and displays them to the user.
+     * Continuously prompts the user to enter a subject name until they choose to exit.
+     *
+     * @param masterStudentList The list of all students.
+     * @param in                The scanner object to read user input.
+     */
+    private static void findStudentsWithSubject(ArrayList<Student> masterStudentList, Scanner in) {
         while (true) {
             System.out.println(ENTER_THE_SUBJECT_NAME_TYPE_EXIT_TO_GO_BACK);
-            String subject = in.nextLine().trim();
+            String input = in.nextLine().trim();
 
-            if (subject.equalsIgnoreCase(EXIT)) {
+            if (input.equalsIgnoreCase(EXIT)) {
                 System.out.println(EXITED_THE_COMMAND);
                 Ui.printDivider();
-                break;
+                return;
             }
 
-            viewStudentsBySubject(masterStudentList, subject);
+            if (!input.isEmpty()) {
+                viewStudentsBySubject(masterStudentList, input);
+                return;
+            } else {
```

Comment:
Trivial point - unnecessary else statement unless you plan to include other checks without returning.